packages/appium-tizen-tv-driver/lib/driver.js

```diff
@@ -255,12 +269,37 @@ class TizenTVDriver extends BaseDriver {
 
     const caps = /** @type {StrictTizenTVDriverCaps} */ (tempCaps);
 
+    // Raise an error if the `sdb capabilities` might raise an exception
+    const deviceCaps = await deviceCapabilities({udid: this.opts.udid});
+    // @ts-ignore
+    this.#platformVersion = deviceCaps?.platform_version || DEFAULT_PLATFORM_VERSION;
+    // @ts-ignore
+    if (deviceCaps?.platform_version) {
+      log.info(`The ${this.opts.udid} Tizen platform version is ${this.#platformVersion}`);
+    } else {
+      log.info(`The ${this.opts.udid} Tizen platform version is unknown. Using ${DEFAULT_PLATFORM_VERSION}.`);
+    }
+
     if (caps.rcOnly && caps.rcMode !== RC_MODE_REMOTE) {
       log.info(`The rcMode capability was not set to remote but we are in rcOnly mode, so ` +
                `forcing it to remote`);
       caps.rcMode = this.opts.rcMode = RC_MODE_REMOTE;
     }
 
+    if (caps.rcMode !== RC_MODE_REMOTE && util.compareVersions(this.#platformVersion, '<', '3')) {
+    }
+
+    if (util.compareVersions(this.#platformVersion, '<', '3')) {
+      if (caps.rcMode === RC_MODE_REMOTE) {
+        log.info(`The ${this.opts.udid} Tizen platform version support only rcOnly mode.`);
+        caps.rcOnly = true;
+      } else {
+        throw new errors.SessionNotCreatedError(
+          `Tizen ${this.#platformVersion} needs to be 'remote' mode for 'rcMode'.`
+        );
```

Comment:
e.g.

```
Selenium::WebDriver::Error::SessionNotCreatedError: A new session could not be created. Details: Tizen 2.4.0 needs to be 'remote' mode for 'rcMode'.
from SessionNotCreatedError: A new session could not be created. Details: Tizen 2.4.0 needs to be 'remote' mode for 'rcMode'.
[57] pry(main)> 
```

Reply:
I chosen raising an error method for non-remote mode instead of forcefully making this case `remote` to avoid too aggressive behavior